test/ex_doc/formatter/html_test.exs

```diff
@@ -592,6 +592,41 @@ defmodule ExDoc.Formatter.HTMLTest do
              ] = Jason.decode!(content)["extras"]
     end
 
+    test "custom search data is added to the sidebar", %{tmp_dir: tmp_dir} = context do
+      generate_docs(
+        doc_config(context,
+          source_beam: "unknown",
+          extras: [
+            {"test/fixtures/README.md",
+             search_data: [
+               %{
+                 anchor: "heading-without-content",
```

Comment:
There is one particular annoyance here. The top item of the search data is the "module body" of the page, and it doesn't have an anchor. So we need to allow anchor to be an empty string and add a test that we emit the proper code (i.e. no trailing #).

Reply:
Alternatively, assume that the top-level one is always generated. This will impact other places, please see comment above.